spec/controllers/activity_comments_controller_spec.rb

```diff
@@ -0,0 +1,279 @@
+        let(:user) { create(:partner_organisation_user) }
+
+        it "responds with a 401" do
+          get :edit, params: {activity_id: project_activity.id, id: existing_project_activity_comment.id}
+
+          expect(response.status).to eq(401)
+        end
+      end
+    end
+  end
+
+  describe "#update" do
+    context "when the activity is a programme" do
+      context "when signed in as a BEIS user" do
+        let(:user) { beis_user }
+
+        it "updates the comment and redirects to organisation activity comments path" do
+          put_comment(comment_id: existing_programme_activity_comment.id, activity_id: programme_activity.id)
+
+          updated_comment = Comment.find(existing_programme_activity_comment.id)
+
+          expect(updated_comment.body).to eq(updated_comment_body)
+          expect(response).to redirect_to(organisation_activity_comments_path(beis_user.organisation, programme_activity))
+        end
+      end
+
+      context "when signed in as a partner organisation user" do
+        let(:user) { partner_organisation_user }
+
+        it "responds with a 401" do
+          put_comment(comment_id: existing_programme_activity_comment.id, activity_id: programme_activity.id)
+
+          expect(response.status).to eq(401)
+        end
+      end
+    end
+
+    context "when the activity is a project" do
+      context "when signed in as a BEIS user" do
+        let(:user) { beis_user }
+
+        it "responds with a 401" do
+          put_comment(comment_id: existing_project_activity_comment.id, activity_id: project_activity.id)
+
+          expect(response.status).to eq(401)
+        end
+      end
+
+      context "when signed in as a partner organisation user whose organisation is the same as the activity's report" do
+        let(:user) { partner_organisation_user }
+
+        it "updates the comment and redirects to organisation activity comments path" do
+          put_comment(comment_id: existing_project_activity_comment.id, activity_id: project_activity.id)
+
+          updated_comment = Comment.find(existing_project_activity_comment.id)
+
+          expect(updated_comment.body).to eq(updated_comment_body)
+          expect(response).to redirect_to(organisation_activity_comments_path(partner_organisation_user.organisation, project_activity))
+        end
+      end
+
+      context "when signed in as a partner organisation user whose organisation is different to the activity's report" do
+        let(:user) { create(:partner_organisation_user) }
+
+        it "responds with a 401" do
+          put_comment(comment_id: existing_project_activity_comment.id, activity_id: project_activity.id)
+
+          expect(response.status).to eq(401)
+        end
+      end
+    end
+  end
+
+  def post_comment(activity_id:, report_id:)
+    post :create, params: {
+      activity_id: activity_id,
+      comment: {body: "Nihilism slips on a banana peel.", report_id: report_id}
```

Comment:
nice comment

Reply:
I used these nonsense generators for the comments 😅
- https://nonsense.x2d.org
- https://generatorfun.com/nonsense-generator